code/game/objects/structures/teleporter.dm

```diff
@@ -141,16 +153,25 @@
 	///The optional cell to power the teleporter if off the grid
 	var/obj/item/cell/cell
 	COOLDOWN_DECLARE(teleport_cooldown)
+	
+	///tag to avoid mixups. Purely for players, not for systems.
+	var/tele_tag
```

Comment:
this can create doubled tags just use a static var and ++ it when making a new one

Reply:
Done